cabal-install/Distribution/Client/CmdLegacy.hs

```diff
@@ -0,0 +1,67 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE RecordWildCards #-}
+{-# LANGUAGE ViewPatterns #-}
+module Distribution.Client.CmdLegacy (legacyCmd) where
+
+import Prelude ()
+import Distribution.Client.Compat.Prelude
+
+import Distribution.Simple.Command
+import Distribution.Simple.Utils
+    ( warn )
+import Distribution.Verbosity 
+    ( normal )
+
+import qualified Data.Text as T
+
+-- Duplicated code, but makes things cleaner and lets me keep how this happens a dirty little secret.
+makeCmd :: CommandUI flags -> (flags -> [String] -> action) -> CommandSpec action
+makeCmd ui action = CommandSpec ui ((flip commandAddAction) action) NormalCommand
+
+deprecationNote :: String
+deprecationNote = 
+    "This command is a part of the legacy v1 style of cabal usage.\n\n" ++
```

Comment:
Rather than saying "This command", better to name the command being used, to save digging in complicated scripts.